text/src/autogluon/text/text_prediction/mx/models.py

```diff
@@ -417,27 +429,37 @@ def train_function(args, reporter, train_df_path, tuning_df_path,
     logger.log(10, cfg)
 
     # Load backbone model
-    backbone_model_cls, backbone_cfg, tokenizer, backbone_params_path, _ \
-        = get_backbone(cfg.model.backbone.name)
     if 'roberta' in cfg.model.backbone.name:
+        backbone_model_cls, backbone_cfg, tokenizer, backbone_params_path, _ \
+            = get_backbone(cfg.model.backbone.name)
         text_backbone = backbone_model_cls.from_cfg(backbone_cfg, return_all_hiddens=True)
     else:
+        backbone_model_cls, backbone_cfg, tokenizer, backbone_params_path, _ \
+            = get_backbone(cfg.model.backbone.name, load_backbone=not continue_training)
         text_backbone = backbone_model_cls.from_cfg(backbone_cfg)
+
     # Build Preprocessor + Preprocess the training dataset + Inference problem type
     # TODO Dynamically cache the preprocessor that has been fitted.
-    if problem_type == MULTICLASS or problem_type == BINARY:
-        label_generator = LabelEncoder()
-        label_generator.fit(pd.concat([train_data[label_column], tuning_data[label_column]]))
-    else:
-        label_generator = None
-    preprocessor = MultiModalTextFeatureProcessor(column_types=column_types,
-                                                  label_column=label_column,
-                                                  tokenizer_name=cfg.model.backbone.name,
-                                                  label_generator=label_generator,
-                                                  cfg=cfg.preprocessing)
-    logger.info('Fitting and transforming the train data...')
-    train_dataset = preprocessor.fit_transform(train_data[feature_columns],
+    if continue_training:
+        with open(preprocessor_path, 'rb') as in_f:
```

Comment:
maybe we want to print some statement here like:  "Continuing training of existing model"

Reply:
Added a logging message.